lib/attributor/hash_dsl_compiler.rb

```diff
@@ -0,0 +1,124 @@
+require_relative 'dsl_compiler'
+
+
+module Attributor
+
+  class HashDSLCompiler < DSLCompiler
+
+    class Requirement
```

Comment:
Can we apply the [Boy-Scout Principle](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule) and not merge more undocumented code? i.e. at least new classes could have some docs on them?